targets/targets.json

```diff
@@ -83,6 +83,11 @@
     },
     "PSA_DUAL_V7_M_SPE": {
```

Comment:
Do we want to rename "PSA_DUAL_V7_M_SPE" to "PSA_DUAL_CORE_SPE" to match our documentation? Same would go for NSPE

Reply:
The name `PSA_DUAL_CORE_SPE` makes more sense than `PSA_DUAL_V7_M_SPE` as both cores might not be same on the targets. Updated both `targets.json` and document.